dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/RelationshipPersister.java

```diff
@@ -47,23 +48,22 @@
 @Component
 public class RelationshipPersister
     extends AbstractTrackerPersister<Relationship, org.hisp.dhis.relationship.Relationship> {
-  private final TrackerConverterService<Relationship, org.hisp.dhis.relationship.Relationship>
-      relationshipConverter;
 
   public RelationshipPersister(
       ReservedValueService reservedValueService,
-      TrackerConverterService<Relationship, org.hisp.dhis.relationship.Relationship>
-          relationshipConverter,
       TrackedEntityChangeLogService trackedEntityChangeLogService) {
 
     super(reservedValueService, trackedEntityChangeLogService);
-    this.relationshipConverter = relationshipConverter;
   }
 
   @Override
   protected org.hisp.dhis.relationship.Relationship convert(
       TrackerBundle bundle, Relationship trackerDto) {
-    return relationshipConverter.from(bundle.getPreheat(), trackerDto, bundle.getUser());
+    if (bundle.getStrategy(trackerDto) == TrackerImportStrategy.UPDATE) {
+      return null;
```

Comment:
why do we not map the object in case of UPDATE?

Reply:
Relationships cannot be updated.
If this is the case, we are ignoring the object in the payload and send a warning.
Ivo created a bug https://dhis2.atlassian.net/browse/DHIS2-18110 where we should improve the way we set the ignored objects to always provide an error/warning. Here it is happening but it is not enforced by the code.